scripts/survived.py

```
@@ -0,0 +1,330 @@
+    adjudication step in D3b catches it with one `git grep` in about a minute.
+    """
+    blob = git("show", "%s:%s" % (ref, path), allow_fail=True)
+    pool = {}
+    if blob:
+        for i, l in enumerate(blob.split("\n"), 1):
+            s = l.strip()
+            if s:
+                pool.setdefault(s, []).append(str(i))
+
+    kept = 0
+    for l in lines:
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)